src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -33,17 +33,15 @@ public class ModuleIntrinsics
         [TraceSource("Modules", "Module loading and analysis")]
```

Comment:
@iSazonov, your last commit had 2 failures in `PowerShell-CI-macos`
Remove-Module core module on module path by name.should be able to remove a module with using Name switch
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Remove-Module.Tests.ps1: line 403
403:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```
Remove-Module core module on module path by name.should be able to remove a module with using ModuleInfo switch
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Remove-Module.Tests.ps1: line 410
410:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```